File: snobot_sim_java/src/main/java/com/snobot/simulator/simulator_components/smart_sc/SmartScEncoder.java
package com.snobot.simulator.simulator_components.smart_sc;

import com.snobot.simulator.module_wrapper.BaseEncoderWrapper;

public class SmartScEncoder extends BaseEncoderWrapper
{

    public SmartScEncoder(int aPort)
    {
        super("CAN Encoder (" + (aPort - BaseCanSmartSpeedController.sCAN_SC_OFFSET) + ")");
    }
}

File: snobot_sim/src/main/native/cpp/SnobotSim/SimulatorComponents/SmartSC/SmartSCEncoder.cpp
#include "SnobotSim/SimulatorComponents/SmartSC/SmartSCEncoder.h"
#include "SnobotSim/SimulatorComponents/SmartSC/CanIdOffset.h"


const std::string SmartSCEncoder::TYPE = "com.snobot.simulator.simulator_components.smart_sc.SmartScEncoder";

SmartSCEncoder::SmartSCEncoder(int aPort) :
        BaseEncoderWrapper("CAN Encoder (" + std::to_string(aPort - CAN_SC_OFFSET) + ")")
{

}

SmartSCEncoder::~SmartSCEncoder()
{

}
